File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private Integer smallestNumber;
    
    public int kthSmallest(TreeNode root, int k) {
        this.smallestNumber = null;
        bfs(root, k, 0);
        return smallestNumber;
    }

    private int bfs(TreeNode currentNode, int k, int currentCount){
        
        if(currentNode == null)
            return currentCount;   
        int leftValue = bfs(currentNode.left, k, currentCount);
        if(leftValue + 1 == k){
            this.smallestNumber = currentNode.val;
        }
            
        int rightValue = bfs(currentNode.right, k, leftValue + 1);
        return rightValue;    
    }
}

File: Data Structures & Algorithms/same-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    

    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        p_sign = {}
        q_sign = {}
        
        self.solve(p, p_sign, 0)
        self.solve(q, q_sign, 0)

        return p_sign == q_sign

    def solve(self, node: Optional[TreeNode], sign: Dict[int, int], index: int) -> None:
        if not node:
            return

        sign[index] = node.val

        self.solve(node.left, sign, (2 * index) + 1)
        self.solve(node.right, sign, (2 * index) + 2)

        